sql-native/src/main/java/org/ballerinalang/sql/nativeimpl/QueryProcessor.java

```diff
@@ -96,7 +95,7 @@ public static BStream nativeQuery(
                 resultSet = statement.executeQuery();
                 List<ColumnDefinition> columnDefinitions;
                 StructureType streamConstraint;
-                if (recordType == null) {
+                if (((BTypedesc) recordType).getDescribingType().getName().equals(Constants.DEFAULT_RECORD_NAME)) {
```

Comment:
@daneshk @niveathika Is this the best way to differentiate the user defined Records from default Record type?

Reply:
Lets support nil, this way, if recordType is null in the native code, we can create a default type and assign it.

```suggestion
                if (recordType == null) {
                    recordType = TypeCreator.createStreamType(...)
```

Reply:
AFAIU, Since now we defined the  second input parameter from `typedesc<record {}>? rowType = ()` to `typedesc<record {}> rowType = <>`  in query remote method. we can not check the null in that case.

Reply:
In this case, 
```
recordType == TypeCreator.createStreamType(...)
```

Reply:
No.What I mean was that we can not check the `recordType == null` here now. because every time it gives `not null` value since the default `rowType` value is set to `<>`

Reply:
From looking into it, it seems we have to use the original name match since the pkg will be different when we use this on different ballerina packages.